heft-plugins/heft-storybook-plugin/src/StorybookPlugin.ts

```diff
@@ -79,79 +83,74 @@ export class StorybookPlugin implements IHeftPlugin<IStorybookPluginOptions> {
    * Generate typings for Sass files before TypeScript compilation.
    */
   public apply(
-    heftSession: HeftSession,
+    taskSession: IHeftTaskSession,
     heftConfiguration: HeftConfiguration,
     options: IStorybookPluginOptions
   ): void {
-    this._logger = heftSession.requestScopedLogger(TASK_NAME);
+    const storybookParameter: CommandLineFlagParameter = taskSession.parametersByLongName.get(
+      '--storybook'
+    ) as CommandLineFlagParameter;
 
-    if (!options.storykitPackageName) {
-      throw new Error(
-        `The ${TASK_NAME} task cannot start because the "storykitPackageName"` +
-          ` plugin option was not specified`
-      );
-    }
+    this._logger = taskSession.logger;
+    this._storykitPackageName = options.storykitPackageName;
+    this._startupModulePath = options.startupModulePath;
 
     const parseResult: IParsedPackageNameOrError = PackageName.tryParse(options.storykitPackageName);
     if (parseResult.error) {
       throw new Error(
-        `The ${TASK_NAME} task cannot start because the "storykitPackageName"` +
+        `The ${taskSession.taskName} task cannot start because the "storykitPackageName"` +
           ` plugin option is not a valid package name: ` +
           parseResult.error
       );
     }
-    this._storykitPackageName = options.storykitPackageName;
 
-    if (!options.startupModulePath) {
-      throw new Error(
-        `The ${TASK_NAME} task cannot start because the "startupModulePath"` +
-          ` plugin option was not specified`
+    // Only tap if the --storybook flag is present.
+    if (storybookParameter.value) {
+      taskSession.requestAccessToPluginByName(
+        '@rushstack/heft-webpack4-plugin',
+        Webpack4PluginName,
```

Comment:
Same as serverless stack plugin, you should be able to put these in consts and to a typecheck-time validation of their values.